src/mscorlib/corefx/System/Globalization/CalendarData.Unix.cs

```diff
@@ -1,36 +1,65 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
-using System;
-using System.Runtime.InteropServices;
-using System.Runtime.CompilerServices;
-using System.Diagnostics.Contracts;
+
 using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
 
 namespace System.Globalization
 {
+    // needs to be kept in sync with CalendarDataType in System.Globalization.Native
+    internal enum CalenderDataType
+    {
+        Uninitialized = 0,
+        NativeName = 1,
+        MonthDay = 2,
+        ShortDates = 3,
+        LongDates = 4,
+        YearMonths = 5,
+        DayNames = 6,
+        AbbrevDayNames = 7,
+        MonthNames = 8,
+        AbbrevMonthNames = 9,
+        SuperShortDayNames = 10,
+        MonthGenitiveNames = 11,
+        AbbrevMonthGenitiveNames = 12,
+        EraNames = 13,
+        AbbrevEraNames = 14,
+    }
+
     internal partial class CalendarData
     {
         private bool LoadCalendarDataFromSystem(String localeName, CalendarId calendarId)
         {
-            // TODO: Implement this.
-            return false;
+            bool result = true;
+            result &= GetCalendarInfo(localeName, calendarId, CalenderDataType.NativeName, out this.sNativeName);
```

Comment:
I like the brevity of this callback pattern, but is does thunk an extra time for the reverse pinvoke. It would be interesting to see if there's any measurable perf impact vs stringbuffer + StringBuilderCache